antismash/modules/clusterblast/core.py

```diff
@@ -781,29 +776,50 @@ def check_diamond_db_compatible(database_file: str) -> bool:
         Returns:
             True if the database file is compatible, False otherwise
     """
-    diamond_version = subprocessing.run_diamond_version()
-    # We don't care about patch versions for now
-    major, minor, _ = diamond_version.split(".")
-    compatible_format = VERSION_FORMAT_MAP[".".join([major, minor])]
 
+    with TemporaryDirectory(change=True):
+        dummy_fasta = "dummy.fa"
+        dummy_db = "dummy.dmnd"
+        with open(dummy_fasta, "w") as handle:
+            handle.write(">test\nM\n")
+        subprocessing.run_diamond_makedb(dummy_db, dummy_fasta)
+        compatible_format = _extract_db_format(dummy_db)
+
+    try:
+        db_format = _extract_db_format(database_file)
+    except ValueError:
+        return False
+
+    if db_format != compatible_format:
+        logging.debug("Incompatible database format for %s. Expected %s but found %s.",
+            database_file, compatible_format, db_format
+        )
+        return False
+    return True
+
+
+def _extract_db_format(database_file: str) -> int:
+    """ Extract version from a diamond database file
+
+        Arguments:
+            database_file: the path to the database file to extract the version from
+
+        Returns:
+            The database version as an integer
+    """
     with open(database_file, 'rb') as handle:
         chunk = handle.read(16)
 
     if len(chunk) != 16:
         logging.debug("Database %s appears corrupted.", database_file)
-        return False
+        raise ValueError()
```

Comment:
This should probably have a message so that if it escapes in future it'll be informative, also tests of this function could use it at some point